com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/service/delegates/actions/FindCopybookCommandTest.java

```diff
@@ -0,0 +1,116 @@
+            new Diagnostic(RANGE, "Syntax error", Error, null, null);
+    private static final Diagnostic COPYBOOK_ERROR =
+            new Diagnostic(
+                    RANGE, COPYBOOK_NAME + ": Copybook not found", Error, null, MISSING_COPYBOOK.name());
+
+    /** Test a code action to resolve the required copybook created for the missing copybook error. */
+    @Test
+    public void testCollectCommandsOrActionsPositive() {
+        CodeActionProvider provider = new FindCopybookCommand();
+
+        List<Diagnostic> diagnostics = singletonList(COPYBOOK_ERROR);
+        CodeActionParams params =
+                new CodeActionParams(
+                        new TextDocumentIdentifier(DOCUMENT_URI), RANGE, new CodeActionContext(diagnostics));
+
+        List<Either<Command, CodeAction>> actions = provider.collectCommandsOrActions(params);
+
+        assertEquals(1, actions.size());
+
+        CodeAction action = actions.get(0).getRight();
+        assertCorrectCodeAction(action);
+    }
+
+    /** Test nothing created for a syntax error. */
+    @Test
+    public void testCollectCommandsOrActionsNegative() {
+
+        CodeActionProvider provider = new FindCopybookCommand();
+
+        List<Diagnostic> diagnostics = singletonList(SYNTAX_ERROR);
+        CodeActionParams params =
+                new CodeActionParams(
+                        new TextDocumentIdentifier(DOCUMENT_URI), RANGE, new CodeActionContext(diagnostics));
+
+        List<Either<Command, CodeAction>> actions = provider.collectCommandsOrActions(params);
+
+        assertEquals(0, actions.size());
+    }
+
+    /** Test only correct errors processed. */
+    @Test
+    public void testCollectCommandsOrActionsMixed() {
+        CodeActionProvider provider = new FindCopybookCommand();
+
+        List<Diagnostic> diagnostics = Arrays.asList(SYNTAX_ERROR, COPYBOOK_ERROR);
+        CodeActionParams params =
+                new CodeActionParams(
+                        new TextDocumentIdentifier(DOCUMENT_URI), RANGE, new CodeActionContext(diagnostics));
+
+        List<Either<Command, CodeAction>> actions = provider.collectCommandsOrActions(params);
+
+        assertEquals(1, actions.size());
+
+        CodeAction action = actions.get(0).getRight();
+        assertCorrectCodeAction(action);
+    }
+
+    private void assertCorrectCodeAction(CodeAction action) {
+        assertEquals(TITLE, action.getTitle());
+        assertEquals(QuickFix, action.getKind());
+        assertEquals(COPYBOOK_ERROR, action.getDiagnostics().get(0));
+        assertNull(action.getEdit());
+
+        Command command = action.getCommand();
+        assertEquals(TITLE, command.getTitle());
+        assertEquals(MISSING_COPYBOOK.name(), command.getCommand());
+
+        List<Object> arguments = command.getArguments();
+        assertEquals(2, arguments.size());
+        assertEquals(COPYBOOK_NAME, arguments.get(0));
+        assertEquals(DOCUMENT_URI, arguments.get(1));
+    }
+}
```

Comment:
Newline here! Is it a time to enable static code checks as part of CI?

Reply:
well..I think that an IDE with configured google format enabled and reformat option after save should be enough

Reply:
Why not?